### PR TITLE
fix(cohorts): Alerting functionality when user enters wrong details while creating cohort

### PR DESCRIPTION
**Pull request by sksDonni**
Closes - #18620 
At present, When a user clicks on save cohort button without properly filling details, the site feels stuck. The user has to scroll down to the highlighted error box to rectify the details.

Such a situation can be avoided by a simple toast alerting the user that the details filled out in the form has errors.

https://github.com/PostHog/posthog/assets/53624363/93134ecc-9f5d-443a-a993-4b108878e28a

Testing - 
1. Go to cohorts
2. Click on add criteria group
3. The toast with error message should appear on the bottom right corner

### CODE REVIEW

File: frontend/src/scenes/cohorts/cohortEditLogic.ts [hunk 1]
```diff
@@ -205,6 +206,7 @@ export const cohortEditLogic = kea<cohortEditLogicType>([
                 },
             }),
             submit: (cohort) => {
+                actions.submitCohort()
```
Comment (neilkakkar) [inline frontend/src/scenes/cohorts/cohortEditLogic.ts:209]: this action exists by default thanks to forms, you shouldn't need to add it yourself, was this not working if you remove the action and this line?

[you can check the generated `cohortEditLogicType.ts` file for all existing actions

Comment (sksDonni) [reply frontend/src/scenes/cohorts/cohortEditLogic.ts:209]: Yeah, will check the functionality after removing this line. Will update the PR accordingly if it works

Comment (neilkakkar) [reply frontend/src/scenes/cohorts/cohortEditLogic.ts:209]: this and the line in actions for the same 👍

Comment (sksDonni) [reply frontend/src/scenes/cohorts/cohortEditLogic.ts]: @neilkakkar, I have made the required changes. Can you please take a look!

**Review by neilkakkar — APPROVED** [review]
perfect, thanks @sksDonni

**Comment by sksDonni** [comment]
@neilkakkar, #18614 is very similar to the issue here. Can we solve it in a similar way?

**Comment by neilkakkar** [comment]
indeed! However, we also want to expand the right section in surveys, since it can be pretty hard otherwise to figure out where the error is

**Pull request merged**